### PR TITLE
REF: collect attribute-setting at the end of create_axes

### PR DESCRIPTION
**Pull request by jbrockmendel**
Ideally we wouldn't be altering `self` inplace at all, but we can make this a lot easier to reason about by collecting all of the attribute-setting at the end of the create_axes call.

`self.non_index_axes` I couldn't get all the way to the end of the call without a more invasive edit, which ill save for a separate pass to keep the diff smallish

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -3785,15 +3785,13 @@ def create_axes(
             )
 
         # create according to the new data
-        self.non_index_axes = []
-        self.data_columns = []
+        nia: List = []
```
Comment (WillAyd) [inline pandas/io/pytables.py]: Any chance of adding a subtype here?

Comment (WillAyd) [inline pandas/io/pytables.py]: Any chance of adding a subtype here?

**Review by WillAyd — COMMENTED** [review]
Sure on board with this

File: pandas/io/pytables.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/io/pytables.py]: why the rename? I would imagine `non_index_axes` newcomer friendly than a TLA.

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -3785,15 +3785,13 @@ def create_axes(
...
+        dcs = []
```
Comment (simonjayhawkins) [inline pandas/io/pytables.py]: same

File: pandas/io/pytables.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: > why the rename? I would imagine non_index_axes newcomer friendly than a TLA.

i wanted something not-identical to the self.foo name for grepping purposes, but can try to find a middle ground that is more descriptive

> Any chance of adding a subtype here?

I'll give it the ol' college try.

**Comment by jbrockmendel** [comment]
changed to more descriptive names, added annotation for new_data_columns, non_index_axes i kept as `List`because I think we can do a better job of being consistent about what we're passing, want to wait to annotate until then

**Comment by jreback** [comment]
sure

**Pull request merged**